### PR TITLE
Sketches Schema UI view

### PR DESCRIPTION
**Pull request by skrawcz**
Original description follows. This has been fixed up to work correctly.

This is hacky UI code. It does not handle "run ids".

1. Adds to pandas & pyspark schema capture using `h_schema.py`.
2. Adds SchemaView and related types to try to show a table...

So it's either we've set something up with too many things, or this is just a factor of using typescript... Basically I dont like how many things I had to add to do this..

Otherwise it's non-obvious to me what the pattern should be to handle comparing runs -- this UI change for the schema table doesn't take that into account -- it should probably use Generic Table but I couldn't wrap my head around it.

Screen Shot 2024-07-04 at 3 18 26 PM
Screen Shot 2024-07-04 at 3 17 54 PM
Screen Shot 2024-07-04 at 3 26 52 PM

## Changes
 - SDK
 - DataObservabilityView

## How I tested this
- locally

## Notes
- need help to handle multiple run Ids and styling it appropriately.
- we might not need to use `h_schema` since I kind of already have some of this... we should align on what the "schema" view actually takes in too.

## Checklist

- [ ] PR has an informative and human-readable title (this will be pulled into the release notes)
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code passed the pre-commit check & code is left cleaner/nicer than when first encountered.
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future TODOs are captured in comments
- [ ] Project documentation has been updated if adding/changing functionality.

### CODE REVIEW

**Comment by skrawcz** [comment]
To complete this:
 - decide on how to create schema stuff — do we use pyarrow stuff or not… (it would mean a dependency on pyarrow is required)
 - redo the table view to enable comparing multiple run IDs.
 - fix SDK tests

**Comment by elijahbenizzy** [comment]
Fixed up schema view: 
image

Added expand/contract now that we have multiple: 
image

File: ui/frontend/src/components/dashboard/Runs/Task/result-summaries/DataObservability.tsx [hunk 1]
```diff
@@ -15,76 +15,148 @@ import {
...
+        // dataTypeDisplay={(item: string) => {
+        //   return (
+        //     <RunLink
+        //       projectId={props.projectId}
+        //       runId={parseInt(item) as number}
+        //       setHighlightedRun={() => void 0}
+        //       highlightedRun={null}
+        //     ></RunLink>
+        //   );
+        // }}
```
Comment (skrawcz) [inline ui/frontend/src/components/dashboard/Runs/Task/result-summaries/DataObservability.tsx:159]: commented out?

File: ui/sdk/src/hamilton_sdk/tracking/pyspark_stats.py [hunk 2]
```diff
@@ -73,53 +74,35 @@ def _introspect(df: ps.DataFrame) -> Dict[str, Any]:
...
-            "name": "Cost Explain",
-        },
-        {
-            "observability_type": "primitive",
-            "observability_value": {
-                "type": str(str),
-                "value": o_value["extended_explain"],
-            },
-            "observability_schema_version": "0.0.1",
-            "name": "Extended Explain",
```
Comment (skrawcz) [inline ui/sdk/src/hamilton_sdk/tracking/pyspark_stats.py]: I don't see these added to additional...

**Review by elijahbenizzy — APPROVED** [review]
This has been fixed up + approved

File: ui/sdk/src/hamilton_sdk/tracking/langchain_stats.py [hunk 3]
```diff
@@ -43,7 +43,7 @@ def compute_stats_lc_docs(
     from langchain_core import messages
 
     msg = messages.BaseMessage(content="Hello, World!", type="greeting")
-    print(stats.compute_stats(msg, "greeting", {}))
+    print(data_observation.compute_stats(msg, "greeting", {}))
```
Comment (skrawcz) [inline ui/sdk/src/hamilton_sdk/tracking/langchain_stats.py:46]: got to remove these 😆

**Pull request merged**